### PR TITLE
Generate Config.swift to isolate the project

### PR DESCRIPTION
**Pull request by fortmarek**
### Short description 📝

I am doing a couple of changes for our server acceptance tests:
- After a Tuist remote project is generated, we regenerated the `Tuist/Config.swift` with the new `fullHandle`, so the tests are run in full isolation
- Added a couple of assertions to be shared with our private repo
- Changed `ios_app_with_frameworks` to be a single project instead of a workspace. This makes it easy to work with in our acceptance tests.

### How to test the changes locally 🧐

CI should be green.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistAcceptanceTesting/ServerAcceptanceTestCase.swift [hunk 1]
```diff
@@ -19,6 +20,18 @@ open class ServerAcceptanceTestCase: TuistAcceptanceTestCase {
         try await run(AuthCommand.self, "--email", email, "--password", password)
         try await run(OrganizationCreateCommand.self, organizationHandle)
         try await run(ProjectCreateCommand.self, fullHandle)
+        try FileHandler.shared.write(
+            """
+            import ProjectDescription
+
+            let config = Config(
+                fullHandle: "\(fullHandle)",
+                url: "\(ProcessInfo.processInfo.environment["TUIST_URL"] ?? "https://cloud-canary.tuist.io")"
```
Comment (pepicrft) [inline Sources/TuistAcceptanceTesting/ServerAcceptanceTestCase.swift:29]:
```suggestion
                url: "\(ProcessInfo.processInfo.environment["TUIST_URL"] ?? "https://canary.tuist.io")"
```

File: Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift [hunk 2]
```diff
@@ -12,6 +13,9 @@ public enum Destination {
     case simulator, device
 }
 
+// swiftlint:disable:next force_try
+public let cacheDirectory = try! TemporaryDirectory(removeTreeOnDeinit: true)
```
Comment (pepicrft) [inline Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift:17]: What about scoping temporary directories to individual test cases?

File: Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift [hunk 3]
```diff
@@ -30,15 +34,21 @@ open class TuistAcceptanceTestCase: XCTestCase {
...
             validating: ProcessInfo.processInfo.environment[
                 "TUIST_CONFIG_SRCROOT"
             ]!
         )
 
+        setenv(
+            "XDG_CACHE_HOME",
+            cacheDirectory.path.pathString,
+            1
+        )
```
Comment (pepicrft) [inline Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift:50]: 😅😅 We need to move away from this pattern at some point for the sake of parallelization.

**Review by pepicrft — APPROVED** [review]

File: Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift (hunk 2 above)
Comment (fortmarek) [reply Sources/TuistAcceptanceTesting/TuistAcceptanceTestCase.swift]: I ended up using the temporary directory created with the `MockEnvironment` 👍

**Pull request merged**